app/grandchallenge/algorithms/views.py

```diff
@@ -335,6 +338,82 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
+class AlgorithmImageMarkDesired(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, View
+):
+    permission_required = "algorithms.change_algorithm"
+    raise_exception = True
+
+    @cached_property
+    def algorithm(self) -> Algorithm:
+        return self.algorithm_image.algorithm
+
+    @cached_property
+    def algorithm_image(self) -> AlgorithmImage:
+        return get_object_or_404(AlgorithmImage, pk=self.kwargs["pk"])
+
+    def get_permission_object(self):
+        return self.algorithm
+
+    def post(self, request, *args, **kwargs):
+        if not self.algorithm_image.is_manifest_valid:
+            messages.error(
+                request, "Cannot mark an invalid image as active image."
+            )
+            return HttpResponseRedirect(
+                self.algorithm_image.get_absolute_url()
+            )
+
+        if not self.algorithm_image.can_execute:
+            image_upload_in_progress = (
+                self.algorithm_image.import_status
+                == ImportStatusChoices.STARTED
+            )
+            if not image_upload_in_progress:
+                self.algorithm_image.import_status = (
+                    ImportStatusChoices.STARTED
+                )
+                self.algorithm_image.save()
+                upload_to_registry_and_sagemaker.signature(
+                    kwargs={
+                        "app_label": self.algorithm_image._meta.app_label,
+                        "model_name": self.algorithm_image._meta.model_name,
+                        "pk": self.algorithm_image.pk,
+                        "mark_as_desired": True,
+                    }
+                ).apply_async()
+                messages.success(
+                    request,
+                    "Image validation and upload to registry in progress. "
+                    "It can take a while for this image to become active, please be patient.",
+                )
+            else:
+                messages.warning(
+                    request, "Image updating already in progress."
+                )
+            return HttpResponseRedirect(
+                self.algorithm_image.get_absolute_url()
+            )
+
+        if (
+            self.algorithm_image.is_manifest_valid
+            and self.algorithm_image.can_execute
```

Comment:
Do you think that these guards could be moved to `mark_desired_version` as we always want that to be true.

Reply:
They are already in the method. I can just remove them here actually.